crates/bevy_ecs/src/component.rs

```diff
@@ -177,6 +182,9 @@ pub struct ComponentDescriptor {
     // this descriptor describes.
     // None if the underlying type doesn't need to be dropped
     drop: Option<for<'a> unsafe fn(OwningPtr<'a>)>,
+    // SAFETY: this function must be safe to call with pointers pointing to items of the type
+    // this descriptor describes. Must never panic.
+    swap: for<'a> unsafe fn(Ptr<'a>, Ptr<'a>),
```

Comment:
This is going to be impossible to implement if the type size is dynamically generated. You did have to pass in the Layout in that case.

Reply:
Alternatively, we could implement this swapping behavior by swapping [MaybeUninit<u8>] s instead

Reply:
No longer relevant due to a change in strategy.